docs/relationship_synthesis.md

```diff
@@ -0,0 +1,266 @@
+The mapping of origins to sources allows for performance improvements when evaluating rules via the matching system.
+
+## Conditions
+
+`Conditions` are used to determine if a rule should be applied to a given data point. 
+The proposal for `conditions` is an iterative improvement over the existing rules format to allow for 
+future changes without major breaking updates.
+
+The following example demonstrates various capabilities supported by conditions:
+
+```yaml
+conditions:
+  - attribute: eventType
+    anyOf: ["NginxSample", "K8sPodSample", "K8sContainerSample"]
+  - attribute: host.guid
+    present: yes
+  - attribute: metricName
+    startsWith: "DurationByCaller/"
+    caseSensitive: true
+  - attribute: metricName
+    regex: ".*\.rds\.amazonaws\.com.*"
+```
+
+The `attribute` field specifies the name of the attribute in the data point, and the operation to perform is defined using matchers 
+such as `anyOf`, `present`, `startsWith`, and `regex`. 
+Only one matcher should be present in each condition, and matching is case-insensitive by default unless 
+specified otherwise with `caseSensitive: true`. 
+Multiple conditions must all match for the rule to apply; there is no support for `OR` conditions.
+
+## Relationship
+
+The `relationship` section defines aspects related to the created relationship.
+
+### Expires
+
+The `expires` field allows configuring the duration for which the relationship should exist if it is not reported within that timeframe. 
+By default, this field is set to ***75 minutes*** (the default for the relationship platform). 
+The value for expires follows the `ISO-8601` format and must be between ***10 minutes*** and ***75 minutes***, which is the time interval 
+currently supported by the platform.
+
+Explicitly providing the expires field is recommended for clarity, even though the default is 75 minutes.
+
+### RelationshipType
+
+The `relationshipType` field specifies the type of relationship to create. Only predefined values from the 
+supported relationship types list are accepted.
+
+#### Source & Target Resolvers
+
+Resolvers define how the source and target GUIDs are determined. There are three types of resolvers:
+
+1. `extractGuid`: Extracts the GUID from an existing attribute in the telemetry.
+2. `buildGuid`: Constructs the GUID using various pieces of information from the telemetry.
+3. `lookupGuid`: Performs a lookup using candidates to find the matching GUID based on telemetry attributes. 
+
+##### extractGuid
+
+The `extractGuid` resolver is the simplest and allows specifying an attribute to use as the GUID.
+
+Example:
+
+```yaml
+  source:
+    extractGuid:
+      attribute: host.guid
+      entityType:
+        value: HOST
+  target:
+    extractGuid:
+      attribute: entityGuid
+      entityType:
+        attribute: nr.entityType
+```
+
+If the resolved GUID corresponds to an `INFRA-NA` GUID, the relationship platform needs to know the actual type behind the GUID. 
```

Comment:
We are leaking INFRA-NA here which is an internal concept, not sure what to do about it. cc @naxhh

Reply:
Extracted INFRA-NA to its own section: https://github.com/newrelic/entity-definitions/pull/1138/commits/cfd617ee612b3b7b28edb5afe16b39458cf3cf9a